api/src/test/java/org/openmrs/api/impl/UserServiceImplTest.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.impl;
+
+import org.junit.Test;
+import org.openmrs.User;
+import org.openmrs.api.PasswordException;
+import org.openmrs.api.UserService;
+
+/**
+ * Unit tests for methods that are specific to the {@link UserServiceImpl}. General tests that
+ * would span implementations should go on the {@link UserService}.
+ */
+public class UserServiceImplTest {
+
+	// testing UserServiceImpl.changePassword(String user, String pw)
+	@Test(expected = PasswordException.class)
+	public void changePassword_shouldThrowPassordExceptionWithEmptyStringPassword() throws Exception {
+		User user = new User();
+		user.setUsername("john");
+		user.setSystemId("1");
+		
+		UserService userService = new UserServiceImpl();
+		
+		userService.changePassword(user, "");
+	}
+	
+	// testing UserServiceImpl.changePassword(User user, String hashedPassword, String salt)
+	@Test(expected = PasswordException.class)
+	public void changePassword_shouldThrowPasswordExceptionWithNullPassword() throws Exception {
+		User user = new User();
+		user.setUsername("john");
+		user.setSystemId("1");
+		
+		UserService userService = new UserServiceImpl();
+		userService.changeHashedPassword(user, null, "");
+	}
+	
+	// testing UserServiceImpl.changePassword(String pw, String pw2)
+	@Test(expected = PasswordException.class)
+	public void changePassord_shouldThrowPasswordExceptionWithAlldigitPassword() throws Exception {
+		String newPwd = "12345"; // invalid
+		String oldPwd = "oldpassword";
+		
+		UserService userService = new UserServiceImpl();
+		
+		userService.changePassword(oldPwd, newPwd);
+	}
```

Comment:
All tests work but for this last one that throws API exception when changePassword, calls Context.getAuthenticatedUser(), with the following stack trace
<b>java.lang.Exception: Unexpected exception, expected<org.openmrs.api.PasswordException> but was<org.openmrs.api.APIException>
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:28)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:271)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:238)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:63)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:236)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:53)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:229)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:309)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:86)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:459)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:675)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:382)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:192)
Caused by: org.openmrs.api.APIException: A user context must first be passed to setUserContext()...use Context.openSession() (and closeSession() to prevent memory leaks!) before using the API
	at org.openmrs.api.context.Context.getUserContext(Context.java:245)
	at org.openmrs.api.context.Context.getAuthenticatedUser(Context.java:644)
	at org.openmrs.api.impl.UserServiceImpl.changePassword(UserServiceImpl.java:357)
	at org.openmrs.api.impl.UserServiceImplTest.changePassord_shouldThrowPasswordExceptionWithAlldigitPassword(UserServiceImplTest.java:54)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:47)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:44)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.ExpectException.evaluate(ExpectException.java:19)
	... 15 more

</b>